cpp/src/arrow/filesystem/CMakeLists.txt

```diff
@@ -36,6 +36,15 @@ if(ARROW_S3)
                           ${GFLAGS_LIBRARIES} GTest::gtest)
     add_dependencies(arrow-tests arrow-s3fs-narrative-test)
   endif()
+
+  if(ARROW_BUILD_BENCHMARKS AND ARROW_PARQUET)
+    add_arrow_benchmark(s3fs_benchmark PREFIX "arrow-filesystem")
+    if(ARG_STATIC_LINK_LIBS)
```

Comment:
I don't think `ARG_STATIC_LINK_LIBS` exists at the top level?

Reply:
I've changed this to look at `ARROW_TEST_LINKAGE` instead.